### DISCUSSION TITLE
How can I implement specific controls during the delete?

### DISCUSSION THREAD

**Discussion by bzy107** [filamentphp/filament#9247] [Help] [answered]
In a Resource class, I have DeleteAction within `->actions([]. 
I want to control the deletion in a way that it's not allowed under certain conditions and display an error message. 
How can I achieve this?

**Comment by VincvanAs** [comment, marked as answer]
You could create a custom action where you call a function. Something like:


```
Action::make('deleteCheck')
    ->button()
    ->action(fn () => $this->deleteCheck())

public function deleteCheck()
{
// your logica here
}
```


Another option is ( dont know if this is best practice )  to create a boot function in your model something like:

```
 protected static function boot()
    {
        parent::boot();
        static::deleting(function ($category) {
            if ($category->children->count() > 0) {
                Notification::make()
                    ->title('error')
                    ->danger()
                    ->send();
                return false;
            }    
        });

    }
```

**Reply by bzy107** [reply]
Thank you for your response.

It seems like I can handle this with `->action()`.
```php
DeleteAction::make()->action(fn() => ...)
```
should work.
